src/cpp/fastdds/core/policy/QosPoliciesSerializer.hpp

```diff
@@ -1240,6 +1245,645 @@ inline bool QosPoliciesSerializer<TopicDataQosPolicy>::read_from_cdr_message(
+    valid &= rtps::CDRMessage::addOctet(cdr_message, 0);
+    valid &= rtps::CDRMessage::addOctet(cdr_message, 0);
+
+    return valid;
+}
+
+template<>
+inline bool QosPoliciesSerializer<RTPSReliableWriterQos>::read_content_from_cdr_message(
+        RTPSReliableWriterQos& qos_policy,
+        rtps::CDRMessage_t* cdr_message,
+        const uint16_t parameter_length)
+{
+    // times(32) + disable_positive_acks(12) + disable_heatbeat_piggyback(4)
+    if (parameter_length < 48)
+    {
+        return false;
+    }
+
+    uint32_t frac(0);
+
+    bool valid = rtps::CDRMessage::readInt32(cdr_message,
+                    &qos_policy.times.initial_heartbeat_delay.seconds);
+    valid &= rtps::CDRMessage::readUInt32(cdr_message, &frac);
+    qos_policy.times.initial_heartbeat_delay.fraction(frac);
+    valid &= rtps::CDRMessage::readInt32(cdr_message,
+                    &qos_policy.times.heartbeat_period.seconds);
+    valid &= rtps::CDRMessage::readUInt32(cdr_message, &frac);
+    qos_policy.times.heartbeat_period.fraction(frac);
+    valid &= rtps::CDRMessage::readInt32(cdr_message,
+                    &qos_policy.times.nack_response_delay.seconds);
+    valid &= rtps::CDRMessage::readUInt32(cdr_message, &frac);
+    qos_policy.times.nack_response_delay.fraction(frac);
+    valid &= rtps::CDRMessage::readInt32(cdr_message,
+                    &qos_policy.times.nack_supression_duration.seconds);
+    valid &= rtps::CDRMessage::readUInt32(cdr_message, &frac);
+    qos_policy.times.nack_supression_duration.fraction(frac);
+
+    valid &= rtps::CDRMessage::readOctet(cdr_message,
+                    (fastdds::rtps::octet*) &qos_policy.disable_positive_acks.enabled);
+    cdr_message->pos += 3; //padding
+
+    valid &= rtps::CDRMessage::readInt32(cdr_message,
+                    &qos_policy.disable_positive_acks.duration.seconds);
+    valid &= rtps::CDRMessage::readUInt32(cdr_message, &frac);
+    qos_policy.disable_positive_acks.duration.fraction(frac);
+
+    valid &= rtps::CDRMessage::readOctet(cdr_message,
+                    (fastdds::rtps::octet*)&qos_policy.disable_heartbeat_piggyback);
+    cdr_message->pos += 3; //padding
+
+    return valid;
+}
+
+template<>
+inline uint32_t QosPoliciesSerializer<WriterResourceLimitsQos>::cdr_serialized_size(
+        const WriterResourceLimitsQos&)
+{
+    // p_id + p_length +  2*(uint32_t(4) + uint32_t(4) + uint32_t(4))
+    return 2 + 2 + 24;
+}
+
+template<>
+inline bool QosPoliciesSerializer<WriterResourceLimitsQos>::add_to_cdr_message(
+        const WriterResourceLimitsQos& qos_policy,
+        rtps::CDRMessage_t* cdr_message)
+{
+    bool valid = rtps::CDRMessage::addUInt32(cdr_message, (uint32_t)qos_policy.matched_subscriber_allocation.initial);
+    valid &= rtps::CDRMessage::addUInt32(cdr_message, (uint32_t)qos_policy.matched_subscriber_allocation.maximum);
+    valid &= rtps::CDRMessage::addUInt32(cdr_message, (uint32_t)qos_policy.matched_subscriber_allocation.increment);
```

Comment:
Would be nice to factor-out methods for `ResourceLimitedContainerConfig`

Reply:
I will add methods for de/serialize it